### DISCUSSION TITLE
Last Change Entity on HM and HMIP Devices in HA

### DISCUSSION THREAD

**Discussion by SeriousPat** [SukramJ/aiohomematic#1083] [Ideas]
(EN) Is ist possible to get entitys for the last change on the Devices?
(DE) Ist es möglich bei den Geräten die Entität der letzten Änderung mit rein zu nehmen? Diese Entität währe ein 1A Trigger für diverse Automationen wie z.B. "Trigger Letzte Änderung has changed"

![image](https://github.com/danielperna84/hahomematic/assets/98743842/cbc9c67f-c654-41c8-bd04-585ef26d1cf3)
![image](https://github.com/danielperna84/hahomematic/assets/98743842/00125b22-4518-4ef8-9a4f-9a41d12b4c9d)

**Comment by danielperna84** [comment]
Wann sich der Status geändert hat, ist doch schon jetzt in HA zu sehen. Zwar nicht der Wert aus der CCU, aber eben der Zeitpunkt, zu dem HA die letzte Änderung wahrgenommen hat. Was praktisch identisch sein sollte.

Unabhängig davon wird der Zeitstempel sowieso nicht per XML-RPC an HA versendet. Die Daten sind also gar nicht verfügbar.

**Comment by SeriousPat** [comment]
Hi, eben nicht bei allen Geräten.
Beim Homematic Keypad (HmIP-WKP) erkennt man z.B. eine richtige Eingabe anhand der CodeID. Wenn man allerdings 2 oder mehr mal die richtige eingabe macht, ändert er nicht den wert. bzw HA bekommt das nicht mit. Hab auch nixbeim sniffen der Events gesehen.
![image](https://github.com/danielperna84/hahomematic/assets/98743842/b45123e9-0d87-4727-abd5-cebe3de2a541)
Hier habe ich zwischen 10 und 17 Uhr 3-4 mal einen richtigen Code angegeben.
Bedeutet, wenn ich 2 mal hintereinander den richtigen Code angebe, bekommt HomeAssistant nur das erste mal mit. Ist also eigentlich nen Problem vom Gerät. Deswegen wollte auf das Attribut "Letzte Änderung" gehen. Habe ich damals gern bei fhem genutzt.

Wenn das Attribut nicht gesendet/ausgelesen wird. Kann der Thread zu gemacht werden.

**Reply by emufan** [reply]
Hast Du aber auch mal die Werte selbst angeguckt? An dem Graph dort kannst Du auch nicht sehen, wenn der Wert gleich bleibt.

Ich meine Developertools

![image](https://github.com/danielperna84/hahomematic/assets/22775515/7e7d614a-12a2-4bf5-a115-970655a1785b)

**Comment by danielperna84** [comment]
Es wäre vielleicht noch hilfreich, wenn du die Gerätedaten ausliest und hier zur Verfügung stellst: https://github.com/danielperna84/pydevccu

Vielleicht hat der HmIP-WKP ja irgendeinen besonderen Parameter, der hier nützlich sein kann.

**Reply by SeriousPat** [reply]
Dann brauche ich paar tips wie das geht...
Sollen wir das dann nicht hier machen?: https://github.com/danielperna84/hahomematic/discussions/1082

**Reply by SukramJ** [reply]
In den letzten 4 Monaten hat dat schon jemand gemacht.
Ein Parameter "last change", oder ähnlich, wird nicht von der CCU gesendet. [Siehe](https://github.com/danielperna84/pydevccu/blob/master/pydevccu/paramset_descriptions/HmIP-WKP.json)

**Reply by SeriousPat** [reply]
Cool, dann wird man bestimmt bald einfach die Kanalberechtigung einund aus schalten können. Damit bräuchte ich das last change nicht mehr. Siehe: https://github.com/danielperna84/hahomematic/discussions/1082

**Reply by SukramJ** [reply]
> Cool, dann wird man bestimmt bald einfach die Kanalberechtigung ein und aus schalten können.

Und woher nimmst Du dein Wissen?

**Reply by danielperna84** [reply]
> Kanalberechtigung einund aus schalten können

So rein vom Namen her würde ich darauf tippen, dass man das mit dem [ACCESS_AUTHORIZATION](https://github.com/danielperna84/pydevccu/blob/0b52bb01bd5f7c6da6c70a7c8e48b715d670c44d/pydevccu/paramset_descriptions/HmIP-WKP.json#L651) parameter machen kann.

**Reply by SukramJ** [reply]
[ACCESS_AUTHORIZATION](https://github.com/danielperna84/pydevccu/blob/0b52bb01bd5f7c6da6c70a7c8e48b715d670c44d/pydevccu/paramset_descriptions/HmIP-WKP.json#L651) ist nur schreibar (`"OPERATIONS": 2`), sendet keine Events und kann nicht gelesen werden.
Daher können wir dazu auch keine sinnvolle Entität anzeigen.

Man kann aber z.Bsp einen Taster erstellen (über die UI als Helfer), und reagiert darauf in einer Automation. Als Aktion ruft man den Service  [homematicip_local.set_device_value](https://github.com/danielperna84/custom_homematic#examples-in-yaml) auf.
Der Parameter ACCESS_AUTHORIZATION findet man in den Kanälen 1 bis 14.
Als Wert verwendet man entweder 0 und 1 oder `"DISABLE"` und `"ENABLE"`.

Eigentlich macht man alles über die UI und musst nicht YAML verwenden.

**Reply by SeriousPat** [reply]
Hi, ja sorry, hab von fhem auf HA gewechselt, bin da noch etwas verwöhnt. Ich dneke ich mach das mit dem umweg über die virtuellen taster
